src/sbvr-api/hooks.ts

```diff
@@ -351,16 +361,30 @@ const defineApi = (args: HookArgs) => {
 
 export const runHooks = async <T extends keyof Hooks>(
 	hookName: T,
-	hooksList: InstantiatedHooks | undefined,
+	hooksList: Array<[string, InstantiatedHooks]> | undefined,
 	args: Omit<Parameters<NonNullable<Hooks[T]>>[0], 'api'>,
 ) => {
 	if (hooksList == null) {
 		return;
 	}
-	const hooks = hooksList[hookName];
-	if (hooks == null || hooks.length === 0) {
+	const hooks = hooksList
+		.map(([version, $hooks]): [string, InstantiatedHooks[T] | undefined] => [
+			version,
+			$hooks[hookName],
+		])
+		.filter(
+			(v): v is [string, InstantiatedHooks[T]] =>
+				v[1] != null && v[1].length > 0,
+		);
+	if (hooks.length === 0) {
 		return;
 	}
+	if (['POSTRUN', 'PRERESPOND', 'POSTRUN-ERROR'].includes(hookName)) {
+		// Any hooks after we "run" the query are executed in reverse order from newest to oldest
+		// as they'll be translating the query results from "latest" backwards to the version that
+		// was actually requested
+		hooks.reverse();
```

Comment:
It's required for the hooks to be passed in using the correct order so that we can run them in that order, if you pass them in using the wrong order then user expectations would break and generally things would be problematic. So I guess in short the answer is it's guaranteed because it needs to be for the correct behavior and hence the code that creates the hook list ensures it does it in order so that things can work.
So I guess the answer is it's a convention that needs to be followed, after all this function would not have the info necessary to check and re-order the hooks if they are out of order

Reply:
So `version` is also a freeform key and does not follow any syntax? thus, the user of the translations has to ensure the order of registering translations to the pine core, right?

Reply:
`version` is the name of the relevant model, I've updated the terminology so hopefully it's clearer